Spigot-Server-Patches/0496-Allow-delegation-to-vanilla-chunk-gen.patch

```diff
@@ -0,0 +1,71 @@
+From 9a92b34331e5f08498239aa29b6bba768fdba252 Mon Sep 17 00:00:00 2001
+From: MiniDigger <admin@minidigger.me>
+Date: Wed, 29 Apr 2020 02:10:32 +0200
+Subject: [PATCH] Allow delegation to vanilla chunk gen
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index f49193d9d..c7e41c7e3 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -1857,6 +1857,31 @@ public final class CraftServer implements Server {
+         return new CraftChunkData(world);
+     }
+ 
++    // Paper start
++    @Override
++    public ChunkGenerator.ChunkData createVanillaChunkData(World world, int x, int z) {
++        // get empty object
++        CraftChunkData data = (CraftChunkData) createChunkData(world);
++        // do bunch of vanilla shit
++        net.minecraft.server.WorldServer nmsWorld = ((CraftWorld) world).getHandle();
++        net.minecraft.server.ProtoChunk protoChunk = new net.minecraft.server.ProtoChunk(new net.minecraft.server.ChunkCoordIntPair(x, z), net.minecraft.server.ChunkConverter.a); // TODO obfhelper
++        List<net.minecraft.server.IChunkAccess> list = new ArrayList<>();
++        list.add(protoChunk);
++        net.minecraft.server.RegionLimitedWorldAccess genRegion = new net.minecraft.server.RegionLimitedWorldAccess(nmsWorld, list);
++        // call vanilla generator, one feature after another. Order here is important!
++        net.minecraft.server.ChunkGenerator<?> chunkGenerator = ((CraftWorld) world).getHandle().worldProvider.getChunkGenerator();
++        chunkGenerator.createBiomes(protoChunk);
++        chunkGenerator.buildNoise(genRegion, protoChunk);
++        chunkGenerator.buildBase(genRegion, protoChunk);
++        chunkGenerator.doCarving(genRegion.d(), protoChunk,  net.minecraft.server.WorldGenStage.Features.AIR);
```

Comment:
these doCarving calls need to be removed, thats by spigots generator.shouldGenerateCaves() already

Reply:
Any update on this @MiniDigger ? I gathered this was a 'dont pull' comment

Reply:
Come on man! We're all eager to try it out and get it merged in! @MiniDigger